src/Microsoft.IdentityModel.JsonWebTokens/Experimental/JsonWebTokenHandler.ValidateSignature.cs

```diff
@@ -154,68 +159,91 @@ private static ValidationResult<SecurityKey, ValidationError> ValidateSignatureU
                     callContext);
 
                 if (result.Succeeded)
+                {
+                    jwtToken.SigningKey = key;
                     return result;
+                }
 
                 if (result.Error is ValidationError validationError)
                 {
+                    stackFrames ??= [];
+                    foreach (StackFrame stackFrame in validationError.StackFrames)
+                        stackFrames.Add(stackFrame);
+
                     exceptionStrings ??= new StringBuilder();
                     keysAttempted ??= new StringBuilder();
                     exceptionStrings.AppendLine(validationError.MessageDetail.Message);
                     keysAttempted.AppendLine(key.ToString());
                 }
             }
 
+            StackFrame currentStackFrame = ValidationError.GetCurrentStackFrame();
+            StackFrame firstStackFrame = (stackFrames == null) ? currentStackFrame! : stackFrames[0];
```

Comment:
```suggestion
            StackFrame firstStackFrame = (stackFrames == null) ? currentStackFrame : stackFrames[0];
```

Is this needed? It looks like GetCurrentStackFrame doesn't return a null possible value?